### PR TITLE
ANDROID-14530 Fix checkbox colors

### PR DESCRIPTION
**Pull request by yamal-alm**
### :goal_net: What's the goal?
CheckBoxInput component is not using correct colors in compose versions

### :construction: How do we do it?
- Fix compose version of CheckBoxInput component according to definition
- Add screenshot tests to both versions of the component (xml and compose)

### ☑️ Checks
- [x] I updated the documentation, including readmes and wikis. If this is a breaking change, update [UPGRADING.md](../UPGRADING.md) to inform users how to proceed. If no updates are necessary, indicate so.
- [x] Tested with dark mode.
- [x] Tested with API 21.

### :test_tube: How can I test this?
- [x] Mistica App QR or download link
- [x] Reviewed by Mistica design team

### CODE REVIEW

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/Inputs.kt [hunk 1]
```diff
@@ -72,6 +72,8 @@ fun Inputs() {
         TextAreaInputSample()
         Title("Check Box input")
         CheckBoxInputSample()
+        DisableCheckBoxInputSample(isChecked = true)
+        DisableCheckBoxInputSample(isChecked = false)
```
Comment (yamal-alm) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/Inputs.kt:76]: adding two more cases to catalog

File: catalog/src/main/res/layout/screen_inputs_catalog.xml [hunk 2]
```diff
@@ -311,6 +311,14 @@
                     app:inputCheckText="This is a disabled CheckBox"
                     app:inputEnabled="false"/>
 
+            <com.telefonica.mistica.input.CheckBoxInput
+                    android:id="@+id/disabled_checkboxInput_unchecked"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    app:inputCheckText="This is a disabled CheckBox"
+                    app:inputChecked="false"
+                    app:inputEnabled="false"/>
```
Comment (yamal-alm) [inline catalog/src/main/res/layout/screen_inputs_catalog.xml:320]: aligning cases in compose and xml catalogs

File: library/src/main/java/com/telefonica/mistica/compose/input/CheckBoxInput.kt [hunk 3]
```diff
@@ -25,13 +28,24 @@ fun CheckBoxInput(
     enabled: Boolean = true,
     onCheckedChange: (Boolean) -> Unit = {},
 ) {
-    Column(modifier = modifier) {
+    Column(
+        modifier = modifier.alpha(enabled)
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/compose/input/CheckBoxInput.kt:32]: global alpha for the whole checkbox input composable. I'm using an util method from common ui package

File: library/src/main/java/com/telefonica/mistica/compose/input/CheckBoxInput.kt [hunk 4]
```diff
@@ -25,13 +28,24 @@ fun CheckBoxInput(
...
                     enabled = enabled,
+                    colors = CheckboxDefaults.colors(
+                        checkedColor = MisticaTheme.colors.controlActivated,
+                        uncheckedColor = MisticaTheme.colors.control,
+                        disabledColor = if (checked) {
+                            MisticaTheme.colors.controlActivated
+                        } else {
+                            MisticaTheme.colors.control
+                        }
+                    )
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/compose/input/CheckBoxInput.kt:48]: this is one of the main changes in this PR, we were relying on default colors

File: library/src/main/java/com/telefonica/mistica/input/CheckBoxInput.kt [hunk 5]
```diff
@@ -85,6 +88,18 @@ class CheckBoxInput @JvmOverloads constructor(
         }
     }
 
+    private fun setButtonTint() {
```
Comment (yamal-alm) [inline library/src/main/java/com/telefonica/mistica/input/CheckBoxInput.kt:91]: another main fix: setting a different color to the checkbox when it is in unchecked state

File: library/src/test/java/com/telefonica/mistica/button/ButtonTest.kt [hunk 6]
```diff
@@ -105,17 +106,6 @@ internal class ButtonTest(
     }
 }
 
-@StyleRes
-private fun Brand.getBaseThemeForBrand(): Int = when (this) {
```
Comment (yamal-alm) [inline library/src/test/java/com/telefonica/mistica/button/ButtonTest.kt:109]: moved to common place

File: library/src/test/java/com/telefonica/mistica/input/CheckBoxInputTest.kt [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+import com.telefonica.mistica.testutils.ScreenshotsTest
+import com.telefonica.mistica.testutils.TestUtils
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.ParameterizedRobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(ParameterizedRobolectricTestRunner::class)
+class CheckBoxInputTest(
```
Comment (yamal-alm) [inline library/src/test/java/com/telefonica/mistica/input/CheckBoxInputTest.kt:27]: xml component test

File: library/src/test/java/com/telefonica/mistica/testutils/ScreenshotUtils.kt [hunk 8]
```diff
@@ -8,7 +8,11 @@ object ScreenshotUtils {
         val componentOrTestName = if(component == null) {
             TestUtils.findRunningTestMethodName()
         } else {
-            "${component}_${style}"
+            var componentName = component
```
Comment (yamal-alm) [inline library/src/test/java/com/telefonica/mistica/testutils/ScreenshotUtils.kt:11]: minor change, do not append style when null in this common method, not every test suite will use a style param

File: library/src/test/java/com/telefonica/mistica/compose/input/CheckBoxInputTest.kt [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        }
+        if (!isEnabled) {
+            screenshotSuffix.append("_disabled")
+        }
+
+        if (checked) {
+            screenshotSuffix.append("_checked")
+        } else {
+            screenshotSuffix.append("_unchecked")
+        }
```
Comment (jeprubio) [inline library/src/test/java/com/telefonica/mistica/compose/input/CheckBoxInputTest.kt:64]: I guess this "__" separator is done on purpose (the same for the traditional views test).

File: library/src/test/java/com/telefonica/mistica/input/CheckBoxInputTest.kt [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+                    checked.flatMap { isChecked ->
+                        errorTexts.map { errorText ->
+                            arrayOf(brand, isEnabled, isChecked, errorText)
+                        }
+                    }
+                }
+            }
+        }
+
+        private val dummyClickableSpain = object : ClickableSpan() { override fun onClick(widget: View) {} }
```
Comment (jeprubio) [inline library/src/test/java/com/telefonica/mistica/input/CheckBoxInputTest.kt:119]: dummy... ok, clickable.... ok, Spain? 🤣

File: library/screenshots/CheckBoxInputXML_TelefonicaBrand__checked.png
Comment (jeprubio) [inline library/screenshots/CheckBoxInputXML_TelefonicaBrand__checked.png:1]: I wonder if the alignment of the checkbox in XML is OK

**Comment by jdelga** [comment]
The text does not seem to be centered with the checkbox. Is this right ?

image

(Also, I would paste the image in the PR description for future reference)

File: library/src/test/java/com/telefonica/mistica/compose/input/CheckBoxInputTest.kt (hunk 9 above)
Comment (yamal-alm) [reply library/src/test/java/com/telefonica/mistica/compose/input/CheckBoxInputTest.kt:64]: not really, I guess there must be some trailing underscore in the common method that builds the component+brand preffix

File: library/src/test/java/com/telefonica/mistica/input/CheckBoxInputTest.kt (hunk 10 above)
Comment (yamal-alm) [reply library/src/test/java/com/telefonica/mistica/input/CheckBoxInputTest.kt:119]: lol xD good catch

File: library/screenshots/CheckBoxInputXML_TelefonicaBrand__checked.png
Comment (yamal-alm) [reply library/screenshots/CheckBoxInputXML_TelefonicaBrand__checked.png:1]: yes.. I can create a ticket to review that alignment in another PR

**Comment by yamal-alm** [comment]
> The text does not seem to be centered with the checkbox. Is this right ?
> 
> image
> 
> (Also, I would paste the image in the PR description for future reference)

yep, already spotted by Josep as well, I will create a separated ticket for that

**Review by jeprubio — APPROVED** [review]

**Review by jdelga — APPROVED** [review]

**Review by aweell — APPROVED** [review]

**Pull request merged**